toke.c

```diff
@@ -4470,76 +4482,113 @@ S_intuit_more(pTHX_ char *s, char *e)
     {
         return TRUE;
     }
-    else if (*s == '$') {
-
-        /* First char is dollar; lean very slightly to it being a subscript */
+    else if (*s == '$') { /* If the first char is dollar; lean very slightly to
+                             it being a subscript */
         weight = -1;
     }
-    else {
-        /* Otherwise, lean a little more towards it being a character class */
+    else { /* Otherwise, lean a little more towards it being a character class.
+            */
         weight = 2;
     }
 
+    /* Keep track of how many multiple occurrences of the same character there
+     * are */
     char seen[256];
-    Zero(seen,256,char);
+    Zero(seen, 256, char);
     un_char = 255;
+
+    /* Examine each character in the construct */
     for (; s < send; s++) {
         last_un_char = un_char;
         un_char = (unsigned char)*s;
         switch (*s) {
           case '@':
           case '&':
           case '$':
+
+            /* Each additional occurrence of one of these three strongly
+             * indicates it is a subscript */
             weight -= seen[un_char] * 10;
+
+            /* Following one of these characters, we look to see if there is an
+             * identifier already found in the program by that name.  If so,
+             * strongly suspect this isn't a character class */
             if (isWORDCHAR_lazy_if_safe(s+1, PL_bufend, UTF)) {
                 int len;
                 char tmpbuf[sizeof PL_tokenbuf * 4];
                 scan_ident(s, tmpbuf, sizeof tmpbuf, FALSE);
                 len = (int)strlen(tmpbuf);
-                if (len > 1 && gv_fetchpvn_flags(tmpbuf, len,
-                                                UTF ? SVf_UTF8 : 0, SVt_PV))
+                if (   len > 1
+                    && gv_fetchpvn_flags(tmpbuf,
+                                         len,
+                                         UTF ? SVf_UTF8 : 0,
+                                         SVt_PV))
                     weight -= 100;
-                else
+                else    /* Not a multi-char identifier already known in the
+                           program; is somewhat likely to be a subscript */
                     weight -= 10;
             }
-            else if (*s == '$'
+            else if (   *s == '$'
                      && s[1]
-                     && memCHRs("[#!%*<>()-=",s[1]))
+                     && memCHRs("[#!%*<>()-=", s[1]))
             {
-                if (/*{*/ memCHRs("])} =",s[2]))
+                /* Here we have what could be a punctuation variable.  If the
+                 * next character after it is a closing bracket, it makes it
+                 * quite likely to be that, and hence a subscript.  If it is
+                 * something else, more mildly a subscript */
+                if (/*{*/ memCHRs("])} =", s[2]))
```

Comment:
I don't understand why the comment /*{*/

Reply:
> I don't understand why the comment `/*{*/`

Pretty sure this is to avoid the `}` in the string messing up bracket matching in editors.

Reply:
@hvds response is that it is for editors who bounce between matching brackets and don't know to disregard those in string literals, seems quite plausible.  Thanks